hack/overlays/component-labels.yaml

```
@@ -0,0 +1,30 @@
+# Copyright 2021 VMware
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#@ load("@ytt:overlay", "overlay")
+
+#@ def not_matcher(marg):
+#@   def matcher(*args, **kwargs):
+#@     return not marg(*args, **kwargs)
+#@   end
+#@   return matcher
+#@ end
+
+#@overlay/match by=not_matcher(overlay.subset({"kind":"Namespace"})),expects="1+"
+---
+metadata:
+  #@overlay/match missing_ok=True
+  labels:
+    #@overlay/match missing_ok=True
+    app.kubernetes.io/component: cartographer.carto.run
```

Reading thru best practices - https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/ I'm not sure if component or part-of is the best choice. Either way, they don't appear to be qualified, are there examples of this? Otherwise I would just expect the label to be `cartographer`.

---

I don't have a strong feeling here, but we should be consistent between Cartographer and Cartographer Conventions (and all future components).

---

> _I'm not sure if component or part-of is the best choice_

I think from the perspective of cartographer as an umbrella project, `component` makes sense per the definition of _"the component within the architecture"_, such that `conventions` is a component, and ... `cartographer` is another? maybe cartographer should be more like ... `core` or something? (I think just `cartographer` and `conventions` as the "components" make sense having not heard any of us mentioning "core" very often)

---

from [slack](https://kubernetes.slack.com/archives/C02HKPSEKV1/p1649270158823619?thread_ts=1649263370.179639&cid=C02HKPSEKV1)

```suggestion
    app.kubernetes.io/component: cartographer
```